source/fundamentals/crud/write-operations/insert.txt

```diff
@@ -1,12 +1,141 @@
 =================
-Insert a Document
+Insert Operations
 =================
 
 .. default-domain:: mongodb
 
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
 Overview
 --------
 
-In this section, we show you how to call the write operations to **insert**
-documents from a collection in your MongoDB database.
+In this guide, you will learn how to insert documents into MongoDB. 
+
+You can use MongoDB to retrieve, update and delete information. To
+perform any of those operations, that information, such as user profiles
+and orders, needs to exist in MongoDB. For that information to exist,
+you need to first perform an insert operation.
+
+An insert operation inserts a single or multiple documents in MongoDB
+using the  ``insertOne()``, ``insertMany()`` and ``bulkWrite()``
+methods. 
+
+The following sections focus on ``insertOne()`` and
+``insertMany()``. For information on how to use the ``bulkWrite()``
+method, see our ``Bulk Operations <TODO>`` page. 
+
+Insert a Single Document
+------------------------
+
+When you want to insert a single document, use the ``insertOne()``
+method. If the insertion is successful,  the method returns an
+``InsertOneResult`` instance representing the ``_id`` of the new
+document. 
+
+.. note::
+
+   If you omit the ``_id`` field, the driver automatically generates a
+   unique ``_id`` before inserting the document.
+
+   A unique ``_id`` should look something like this:
+
+   .. code-block:: json
+      :copyable: false
+
+      "_id": { "$oid": "60905ef9865e2105ceb7f7b8" }
+
+Example
+~~~~~~~
+
+The following example creates and inserts a document using the
+``insertOne()`` method: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Insert.java
+   :language: java
+   :dedent:
+   :start-after: begin insertOneExample
+   :end-before: end insertOneExample
+
+Your output should look something like this:
+
+.. code-block:: json
+   :copyable: false
+
+   Inserted a document with the following id: 1
+
+Insert Multiple Documents
+-------------------------
+
+When you want to insert multiple documents, use the ``insertMany()``
+method. If the insertion is successful,  the method returns an
```

Comment:
Tacking on to my comment above, can we add more info? The info about `_id` seems like it should go here rather than in a note after talking about the return. What happens if it errors? Will it write all but the documents with errors?